### PR TITLE
lib_manager: iadk: Fix variable initialization issues

### PR DESCRIPTION
**Pull request by jxstelter**
This PR contains data initialization fixes in library manager and IADK modules.

Fix from mtl-002-drop-stable: [https://github.com/thesofproject/sof/pull/6718](https://github.com/thesofproject/sof/pull/6718)

Signed-off-by: Jaroslaw Stelter [Jaroslaw.Stelter@intel.com](mailto:Jaroslaw.Stelter@intel.com)

### CODE REVIEW

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
@jxstelter fyi - some conflicts.

File: src/library_manager/lib_manager.c [hunk 1]
```diff
@@ -451,8 +453,10 @@ static int lib_manager_dma_init(struct lib_manager_dma_ext *dma_ext, uint32_t dm
 
 static int lib_manager_dma_deinit(struct lib_manager_dma_ext *dma_ext, uint32_t dma_id)
 {
-	dma_release_channel(dma_ext->dma->z_dev, dma_id);
-	dma_put(dma_ext->dma);
+	if (dma_ext->dma->z_dev)
+		dma_release_channel(dma_ext->dma->z_dev, dma_id);
+	if (dma_ext->dma)
+		dma_put(dma_ext->dma);
```
Comment (ranj063) [inline src/library_manager/lib_manager.c:459]: the ordering of the if's above don't quite add up right? if (dma_ext->dma) is false, how can we check for if (dma_ext->dma->z_dev) first? And also, you talk about init issue in the commit message and this change isnt quite related to that at all

Comment (jxstelter) [reply src/library_manager/lib_manager.c:467]: It is true of course. Fixed.

**Review by ranj063 — APPROVED** [review]

**Review by kv2019i — APPROVED** [review]

**Comment by kv2019i** [comment]
Proceeding with merge. The TGL IPC4 alsabat failure is caused by #6709 and is a known issue.

**Pull request merged**